pkg/registry/registry_to_model.go

```diff
@@ -55,6 +56,15 @@ func ObjectsAndPropertiesFromBundle(b *Bundle) ([]string, []property.Property, e
 				return nil, nil, property.ParseError{Idx: i, Typ: p.Type, Err: err}
 			}
 			packageRequiredProps = append(packageRequiredProps, property.MustBuildPackageRequired(v.PackageName, v.Version))
+		case constraints.OLMConstraintType:
+			var v constraints.Constraint
+			if err := json.Unmarshal(p.Value, &v); err != nil {
+				return nil, nil, property.ParseError{Idx: i, Typ: p.Type, Err: err}
+			}
+			otherProps = append(otherProps, property.Property{
+				Type:  p.Type,
+				Value: p.Value,
+			})
```

Comment:
Rather than having special case handling for `olm.constraint` would we lose anything by treating all other `dependency` blobs opaquely?

IIRC, the only reason the existing special case handling exists is because there's translation or FBC-specific validation that we have to do.

```suggestion
		default:
			otherProps = append(otherProps, property.Property{
				Type:  p.Type,
				Value: p.Value,
			})
```

Reply:
I'm fine with this generic approach.